### PR TITLE
more Leaders and Lore

### PR DESCRIPTION
**Pull request by whichwit**
**Leaders:**
- Quartermaster

**Lore:**
- Repair Drones
- Seeker Torpedoes
- Railgun Arrays
- Living Structures

### CODE REVIEW

File: haunt-roll-fail/arcs/game-lore.scala [hunk 1]
```diff
@@ -83,6 +87,59 @@ object LoreExpansion extends Expansion {
...
+                .some(systems)(s => f.at(s).starports./(u => PruneStarportAction(f, x, s, u, then).as(Starport.of(f), Image(prefix + "starport" + f.damaged.has(u).??("-damaged"), styles.qbuilding), "in", s)))
+                .cancel
+
+        case PruneCityAction(f, x, s, u, then) =>
+            f.pay(x)
+
+            val u2 = f.reserve --> Starport.of(f)
+
+            if (f.damaged.has(u))
+                f.damaged :+= u2
```
Comment (haunt-roll-fail) [inline haunt-roll-fail/arcs/game-lore.scala:106]: Probably need to remove `u` too from `damaged`, so it wouldn't be build damaged somewhere else.

File: haunt-roll-fail/arcs/game-common.scala [hunk 2]
```diff
@@ -1105,6 +1156,17 @@ object CommonExpansion extends Expansion {
...
                 .cancel
 
+        case RepairsAction(f, x, r, l, then) =>
+            f.pay(x)
+
+            l.foreach { u =>
+                f.damaged :-= u
+            }
+
+            f.log("repaired", l, "in", r, x)
```
Comment (haunt-roll-fail) [inline haunt-roll-fail/arcs/game-common.scala:1166]: Can use `l.comma` for commas between ship names :)

File: haunt-roll-fail/arcs/game-common.scala [hunk 3]
```diff
@@ -781,9 +784,15 @@ object CommonExpansion extends Expansion {
 
         case BattleSystemAction(f, x, r, then) =>
             Ask(f).group(f, "battles in", r, x)
-                .each(f.rivals.%(_.present(r)))(e => BattleFactionAction(f, x, r, e, then).as(e))
+                .each(f.rivals.%(_.present(r)))(e => BattleHitBeforeAction(f, r, e, BattleFactionAction(f, x, r, e, BattleRepairAfterAction(f, r, e, then))).as(e))
```
Comment (haunt-roll-fail) [inline haunt-roll-fail/arcs/game-common.scala:787]: I can't exactly explain why I feel this way, but I think this nesting here is premature.

Also, I think we need to tunnel `used : $[Effect]` from the start all the way through battle actions.
One reason would be `Signal Breaker`/`Railgun Arrays` interaction - we could check and add `Signal Breaker` to `used` in this `PreBattleAction`, then process `Railgun Arrays`, then use `Signal Breaker` after roll if it was available.

File: haunt-roll-fail/arcs/game-common.scala (hunk 2 above)
Comment (whichwit) [reply haunt-roll-fail/arcs/game-common.scala:1166]: Fixed.

File: haunt-roll-fail/arcs/game-lore.scala (hunk 1 above)
Comment (whichwit) [reply haunt-roll-fail/arcs/game-lore.scala:106]: Fixed.

**Comment by riley611** [comment]
Tool Priest not working as intended with build (summon) action

**Pull request closed without merging**